src/rtree/adjust.jl

```diff
@@ -84,7 +84,7 @@ function _updatembr!(node::Branch, child_ix::Integer, child_oldmbr::Rect,
     mbr_dirty && syncmbr!(node)
 
     if mbr_dirty && hasparent(node)
-        _updatembr!(parent(node), pos_in_parent(node), node_oldmbr, con, force=force)
+        _updatembr!(parent(node)::Branch, pos_in_parent(node), node_oldmbr, con, force=force)
```

Comment:
Is it just for type assertion or it is helping with something?

Reply:
It doesn't remove 

```julia
┌ _updatembr!(node::SpatialIndexing.Branch, child_ix::Integer, child_oldmbr::SpatialIndexing.Rect, con::SpatialIndexing.SubtreeContext; force::Bool) @ SpatialIndexing c:\Users\User\.julia\dev\SpatialIndexing.jl\src\rtree\adjust.jl:87
│ no matching method found `kwcall(::@NamedTuple{force::Bool}, ::typeof(SpatialIndexing._updatembr!), ::Union{Nothing, SpatialIndexing.Branch{T, N, V}} where {T, N, V}, ::Nothing, ::SpatialIndexing.Rect, ::SpatialIndexing.SubtreeContext)` (1/2 union split): Core.kwcall(NamedTuple{(:force,)}(tuple(force::Bool)::Tuple{Bool})::@NamedTuple{force::Bool}, SpatialIndexing._updatembr!, SpatialIndexing.parent(node::SpatialIndexing.Branch)::Union{Nothing, SpatialIndexing.Branch{T, N, V}} where {T, N, V}, SpatialIndexing.pos_in_parent(node::SpatialIndexing.Branch)::Union{Nothing, Int64}, node_oldmbr::SpatialIndexing.Rect, con::SpatialIndexing.SubtreeContext)
```

but it should help the compiler a bit.

Reply:
I have thought that `hasparent()` would always be inlined, so static analysis should go inside and figure out that if the result is true, the value of `parent()` can only be `Branch`.
In fact, maybe the actual compiler is smart enough (you can check with `code_warntype()`), and it's just the `report_package()` false alarm.
Interestingly, there's similar call at L141, but `report_package()` does not complain.

Reply:
I don't know if compiler would eliminate this type check after static analysis, since it should always be true.
If the compiler would keep it, it will introduce some overhead to this frequently called method.
I suggest to revert it, because `report_package()` warning looks like false positive, and if we were to apply such checks consistently, it would have to be done in multiple places.

I personally prefer to avoid type asserts in the code, if possible, as it's against Julia philosophy. :)